src/tv2-common/videoSwitchers/Atem.ts

```diff
@@ -259,22 +259,31 @@ export class Atem extends VideoSwitcherBase {
 				}
 		}
 	}
-	private getUpstreamKeyers(keyers: Keyer[]) {
+	private getUpstreamKeyers(keyers: Keyer[]): TSR.TimelineObjAtemME['content']['me']['upstreamKeyers'] {
 		if (!keyers?.length) {
 			return
 		}
-		return keyers.map((keyer) => ({
-			upstreamKeyerId: keyer.config.Number,
-			onAir: keyer.onAir,
-			mixEffectKeyType: 0,
-			flyEnabled: false,
-			fillSource: keyer.config.Fill,
-			cutSource: keyer.config.Key,
-			maskEnabled: false,
-			lumaSettings: {
-				clip: keyer.config.Clip * 10, // input is percents (0-100), atem uses 1-000
-				gain: keyer.config.Gain * 10 // input is percents (0-100), atem uses 1-000
+		const result: TSR.TimelineObjAtemME['content']['me']['upstreamKeyers'] = []
```

Comment:
Not a fan of this implementation, but I don't have a suggestion for how else you would do it from here.

It seems like TSR using indexes of the array instead of the actual ids is the real issue that needs to be fixed and this is just a bandaid solution for that.
Do we know how extensive it would be to make the change in TSR instead of here (and do we know how much else we would break by doing that)?

Reply:
> It seems like TSR using indexes of the array instead of the actual ids is the real issue that needs to be fixed and this is just a bandaid solution for that.

That's what I thought too, but since NRK is doing it the way I did it here, I thought maybe the problem is deeper and would also require changes in atem-state or atem-connection libraries, which I wanted to avoid